### PR TITLE
Close the ResultSet in JdbcDataConnection.java, PostgresCdcListenBeforeExistsIntegrationTest.java [HZ-2903]

### PR DESCRIPTION
**Pull request by orcunc**
Breaking changes (list specific methods/types/messages):
* API
* client protocol format
* serialized form
* snapshot format

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java [hunk 1]
```diff
@@ -124,18 +124,21 @@ public List<DataConnectionResource> listResources() {
...
-            List<DataConnectionResource> result = new ArrayList<>();
-            while (tables.next()) {
-                String[] name = Stream.of(tables.getString("TABLE_CAT"),
-                                tables.getString("TABLE_SCHEM"),
-                                tables.getString("TABLE_NAME"))
-                        .filter(Objects::nonNull)
-                        .toArray(String[]::new);
-
-                result.add(new DataConnectionResource(OBJECT_TYPE_TABLE, name));
+            try (ResultSet tables = metaData.getTables(null, null, "%", null)) {
```
Comment (frant-hartm) [inline hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java:127]: Maybe merge the two try blocks into one?

**Review by frant-hartm — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java (hunk 1 above)
Comment (orcunc) [reply hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java:127]: DatabaseMetaData is not of type AutoCloseable. It won't compile

Comment (frant-hartm) [reply hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java:127]: I meant the following:
```
try (Connection connection = getConnection();
    ResultSet tables = connection.getMetaData().getTables(null, null, "%", null)) {
```

Comment (orcunc) [reply hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java]: rev : 4a56eaea6902b349ecb5946bb7f2231d4c45f013

**Comment by orcunc** [comment]
run-lab-run

**Review by ldziedziul — APPROVED** [review]

**Pull request merged**